Source/WebCore/html/HTMLScriptElement.cpp

```diff
@@ -87,9 +88,68 @@ void HTMLScriptElement::didFinishInsertingNode()
 }
 
 // https://html.spec.whatwg.org/multipage/scripting.html#dom-script-text
-void HTMLScriptElement::setText(String&& value)
+ExceptionOr<void> HTMLScriptElement::setText(std::variant<RefPtr<TrustedScript>, String> value)
 {
-    setTextContent(WTFMove(value));
+    ExceptionOr<String> stringValueHolder = WTF::switchOn(
```

Comment:
This repeated idiom calls for a helper function. We should pass in the `variant` and the `ASCIILiteral` argument to `trustedTypeCompliantString` and share the code.

Reply:
Ive made a new trustedScriptCompliantStringFromVariant function and an equivalent one for trusted script URL. Let me know what you think.